framework/src/base/Assembly.C

```diff
@@ -2740,7 +2740,11 @@ Assembly::jacobianBlockNeighbor(Moose::DGJacobianType type,
                                 unsigned int jvar,
                                 TagID tag)
 {
-  _jacobian_block_neighbor_used[tag][ivar][jvar] = 1;
+  if (type == Moose::ElementElement)
+    _jacobian_block_used[tag][ivar][jvar] = 1;
+  else
+    _jacobian_block_neighbor_used[tag][ivar][jvar] = 1;
```

Comment:
Was this @fdkong's bug that you guys were working on in the beginning of March?

Reply:
Not really a bug I would say. Just need to use it in a right way.

Reply:
It was definitely a bug.  The only defense you might have had would be a doc comment explaining why the functions don't uniformly/consistently do what their names imply - but their was no such API doc.